pandas/core/window/rolling.py

```diff
@@ -1632,20 +1630,20 @@ def quantile(self, quantile, interpolation="linear", **kwargs):
     """
 
     def cov(self, other=None, pairwise=None, ddof=1, **kwargs):
-        if isinstance(self.window, BaseIndexer):
-            validate_baseindexer_support("cov")
-
         if other is None:
             other = self._selected_obj
             # only default unset
             pairwise = True if pairwise is None else pairwise
         other = self._shallow_copy(other)
 
-        # GH 16058: offset window
-        if self.is_freq_type:
-            window = self.win_freq
+        if isinstance(self.window, BaseIndexer):
```

Comment:
I _think_ this is what `def _get_window` is trying to accomplish. Can the fix not be applied there?

Reply:
@mroeschke 
I did some digging, and it looks like that isn't the intention behind `_get_window`. For example, its output is unconditionally used as input to this function:
```python
def calculate_center_offset(window):
    if not is_integer(window):
        window = len(window)
    return int((window - 1) / 2.0)
```
I think it's meant to only return an int. This is the version that gets called in our code (there is another one in `class Window` that returns an array of window weights, but that's irrelevant for our case):
```python
    def _get_window(self, other=None, win_type: Optional[str] = None) -> int:
        """
        Return window length.

        Parameters
        ----------
        other :
            ignored, exists for compatibility
        win_type :
            ignored, exists for compatibility

        Returns
        -------
        window : int
        """
        if isinstance(self.window, BaseIndexer):
            return self.min_periods or 0
        return self.window
```

In addition, we have `_get_window_indexer`, which can a bit confusing if we allow `_get_window` to return a `BaseIndexer` subclass. Overall, I'd rather leave `_get_window` alone for now.

Does that make sense? I've also added some comments to the code as per @jreback 's suggestion, so hopefully that helps clear things up.

Reply:
Ah okay. Yeah the names of these helper functions can be a little vague. Changes look good.